lib/ViewModels/AnimationViewModel.js

```diff
@@ -119,6 +124,74 @@ function updateForNewTopLayer(animationViewModel) {
     }
 }
 
+/**
+ * Set time to nearest time to specified (may be start or end of time range if time is not in range).
+ * @param {Terria} terria instance to set current time on
+ * @param {JulianDate} date to set
+ * @private
+ */
+function _setTimeIfInRange(terria, timeToSet)
+{
+    if (JulianDate.lessThan(timeToSet, terria.clock.startTime)) {
+        terria.clock.currentTime = terria.clock.startTime;
+    }
+    else if (JulianDate.greaterThan(timeToSet, terria.clock.stopTime)) {
+        terria.clock.currentTime = terria.clock.stopTime;
+
+    } else {
+        terria.clock.currentTime = timeToSet;
+    }
+}
+
+/**
+ * Sets the current time of the clock, so the animation starts at a point which can be user defined.
+ * Valid options in config file are:
+ *     timesliderInitTime: "present"                            // closest to today's date
+ *     timesliderInitTime: "start"                              // start of time range of animation
+ *     timesliderInitTime: "end"                                // end of time range of animation
+ *     timesliderInitTime: An ISO8601 date e.g. "2015-08-08"    // specified date or nearest if date is outside range
+ * @param {AnimationViewModel} view model to set current time on
+ */
+function setClockCurrentTime(animationViewModel)
+{
+    var terria = animationViewModel.terria;
+
+    // This is our default. Start at the nearest instant in time.
+    var now = JulianDate.now();
+    _setTimeIfInRange(terria, now);
+
+    if (defined(terria.timeSeriesStack.topLayer.timesliderInitTime))
+    {
+        var startTime = terria.timeSeriesStack.topLayer.timesliderInitTime;
+        switch(startTime)
+        {
+            case "start":
+                terria.clock.currentTime = terria.clock.startTime;
+                break;
+            case "end":
+                terria.clock.currentTime = terria.clock.stopTime;
+                break;
+            case "present":
+                // Set to present by default.
+                break;
+            default:
+                // Note that if it's not an ISO8601 timestamp, it ends up being set to present.
+                // Is it an ISO8601 timestamp?
+                var timestamp;
+                try {
+                    timestamp = JulianDate.fromIso8601(startTime);
+                }
+                catch (e) {
+                    throw new DeveloperError('No good timesliderInitTime specified in config file: ' + startTime);
```

Comment:
`DeveloperError` should only be thrown when a developer uses an API wrong.  The idea is (and Cesium actually does this!) that all DeveloperErrors and related checks can be removed completely from release builds.  Use `TerriaError` or maybe `RuntimeError`.